Kickstarter-iOS/DataSources/DiscoveryFiltersDataSource.swift

```diff
@@ -66,10 +66,14 @@ internal final class DiscoveryFiltersDataSource: ValueCellDataSource {
              inSection: Section.categoriesLoader.rawValue)
   }
 
-  internal func deleteCategoriesLoaderRow() -> [IndexPath] {
-    self.clearValues(section: Section.categoriesLoader.rawValue)
+  internal func deleteCategoriesLoaderRow() -> [IndexPath]? {
+    if !self[section: Section.categoriesLoader.rawValue].isEmpty {
```

Comment:
It could, yes.

I don't have direct access to `self.values` in the subclass of `ValueCellDataSource` though, could add a function to check if it's empty or of a section exists?